Example/ScanDemo 2/Pods/CML_SCAN/ScanView.swift

```diff
@@ -0,0 +1,284 @@
+        self.shouldVibrateOnScan = beTurnedOn
+    }
+
+    func scan() {
+        self.session = AVCaptureSession()
+        if let _ = self.previewLayer {
+            self.previewLayer.frame = self.bounds
+        }
+        self.provideInputDevice()
+    }
+    
+    func provideInputDevice() {
+        do {
+            
+            self.setCaptureDevice()
+            let input = try AVCaptureDeviceInput(device: captureDevice)
+            self.session.addInput(input)
+            self.ProvideOutputToGetPreviewLayer()
+            self.previewLayer.frame = self.frame
+            self.manageViews()
+        } catch {
+            self.error = "Cannot access Camera. Please, check if camera access is provided in the settings."
+        }
+    }
+
+    func setCaptureDevice() {
+        var captureDevice: AVCaptureDevice?
+        let devices = AVCaptureDevice.devices()
+        for dev in devices! {
+            let device = dev as! AVCaptureDevice
+            if isBackCamera {
+                if device.position == .back {
+                    captureDevice = device
+                }
+            } else {
+                if (device as AnyObject).position == .front {
+                    captureDevice = device
+                }
+            }
+        }
+        self.captureDevice = captureDevice
+    }
+    
+    func ProvideOutputToGetPreviewLayer() {
+        let output = AVCaptureMetadataOutput()
+        output.setMetadataObjectsDelegate(self, queue: DispatchQueue.main)
+        self.session.addOutput(output)
+        output.metadataObjectTypes = self.supportedTypes
+        output.rectOfInterest = interestRect
+        let previewLayer = AVCaptureVideoPreviewLayer(session: self.session)
+        previewLayer?.videoGravity = AVLayerVideoGravityResizeAspectFill
+        self.previewLayer = previewLayer!
+        if isBackCamera == false {
+            self.previewLayer.connection.automaticallyAdjustsVideoMirroring = false
+            self.previewLayer.connection.isVideoMirrored = false
+        }
+        self.session.startRunning()
+    }
+
+    func setSupportesTypes(from types: [Types]) {
+        for type in types {
+            switch type {
+            case .qrCode:
+                self.supportedTypes.append(AVMetadataObjectTypeQRCode)
+            case .ean8:
+                self.supportedTypes.append(AVMetadataObjectTypeEAN8Code)
+            case .ean13:
+                self.supportedTypes.append(AVMetadataObjectTypeEAN13Code)
+            case .code128:
+                self.supportedTypes.append(AVMetadataObjectTypeCode128Code)
+            case .code39:
+                self.supportedTypes.append(AVMetadataObjectTypeCode39Code)
+            case .dataMatrix:
+                self.supportedTypes.append(AVMetadataObjectTypeDataMatrixCode)
+            }
+        }
+    }
+
+    func loadSubViews() {
+        let sideOfInterestRect = min(min(self.bounds.width, self.bounds.height) - 50.0, 300)
```

Comment:
Because the `rectOfInterest` property of `AVCaptureMetaDataOutput` demands a CGRect with origin and size values in the range of 0 to 1.